classes/Product.php

```diff
@@ -3297,14 +3297,19 @@ public static function priceCalculation(
         static $address = null;
         static $context = null;
 
-        if ($address === null) {
-            $address = new Address();
-        }
-
         if ($context == null) {
             $context = Context::getContext()->cloneContext();
         }
 
+        if ($address === null) {
+            if (is_object($context->cart) && $context->cart->{Configuration::get('PS_TAX_ADDRESS_TYPE')} != null) {
+                $id_address = $context->cart->{Configuration::get('PS_TAX_ADDRESS_TYPE')};
+                $address = new Address($id_address);
+            } else {
+                $address = new Address();
+            }
+        }
```

Comment:
This change makes me uneasy because it adds even more implicit contextual behavior to this method, which is already very complex.

The way I see it, if the address has an impact on price, then it should be injected into the calculation and instead of figuring it out from context. Of course, doing that would mean that dependent code would need to be refactored...

Reply:
To much refactoring is needed to do this kind of stuff.
That's why I approved this PR. :disappointed: